pkg/tsdb/cloudwatch/cloudwatch.go

```diff
@@ -170,7 +170,7 @@ func (e *cloudWatchExecutor) QueryData(ctx context.Context, req *backend.QueryDa
 
 	_, fromAlert := req.Headers[ngalertmodels.FromAlertHeaderName]
 	fromExpression := req.GetHTTPHeader(query.HeaderFromExpression) != ""
-	isSyncLogQuery := (fromAlert || fromExpression) && model.QueryMode == logsQueryMode
+	isSyncLogQuery := (fromAlert || fromExpression) && model.QueryMode == logsQueryMode || (model.Type == "" && model.QueryMode == logsQueryMode)
```

Comment:
i think i'm missing what this change has to do with calling super.query, is this an unrelated oversight? What situation does this case represent? How does the user get into this mode?

Reply:
> Public dashboards should work with any data source that has the properties `backend` and `alerting` both set to true in its `plugin.json`
> – https://grafana.com/docs/grafana/latest/dashboards/dashboard-public/#supported-data-sources

public dashboard queries execute _like_ alert queries, i.e. they execute on the backend. so we need to handle them synchronously since the `model.Type` is set _during_ execution on the frontend by the query runner and isn't _saved_ with the query
https://github.com/grafana/grafana/blob/ced9fcb94e979f519e113b79da636c6cb71c9683/public/app/plugins/datasource/cloudwatch/query-runner/CloudWatchLogsQueryRunner.ts#L289-L290

also note that the subtype is also set by the query runner on the frontend, which is another reason we need to run it synchronously since we won't know which step the query is on, i.e. whether it just started, or is polling for results.